modules/fiftyone_degrees/device_detection/hook_raw_auction_request.go

```diff
@@ -0,0 +1,185 @@
+package device_detection
+
+import (
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/v2/hooks/hookexecution"
+	"github.com/prebid/prebid-server/v2/hooks/hookstage"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+)
+
+type deviceMapper interface {
+	HydrateDeviceType(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateUserAgent(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateMake(device hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateModel(payload hookstage.RawAuctionRequestPayload, extMap map[string]any) ([]byte, error)
+	HydrateOS(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateOSVersion(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateScreenHeight(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateScreenWidth(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydratePixelRatio(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateJavascript(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydrateGeoLocation(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+	HydratePPI(payload hookstage.RawAuctionRequestPayload) ([]byte, error)
+}
+
+func handleAuctionRequestHook(ctx hookstage.ModuleInvocationContext, deviceDetector deviceDetector, evidenceExtractor evidenceExtractor) (hookstage.HookResult[hookstage.RawAuctionRequestPayload], error) {
+	var result hookstage.HookResult[hookstage.RawAuctionRequestPayload]
+
+	// If the hook is not enabled, return the result without any changes
+	if ctx.ModuleContext[DDEnabledCtxKey] == nil || ctx.ModuleContext[DDEnabledCtxKey] == false {
+		return result, nil
+	}
+
+	if ctx.ModuleContext == nil || deviceDetector == nil {
+		return result, hookexecution.NewFailure("error getting device detector")
+	}
```

Comment:
Thank you, this moment was missed. I just checked, we did not get a panic and returned on line 18, but overall this code could be improved in this way:
1) I'll leave only the `ctx.ModuleContext == nil` condition. 
2) The `deviceDetector == nil` check does not make sense because it is created in the module builder. So it either exists or the module returns an error.
And the `ctx.ModuleContext[DDEnabledCtxKey] == nil || ctx.ModuleContext[DDEnabledCtxKey] == false` also do not make a lot of sense because we are setting this key only in one place and exactly after creating the ModuleContext. So, ctx.ModuleContext == nil should be enough.
3) As you suggested we could return an error here to the user saying that the entrypoint hook was not configured, because currently we are not handling this case in any way. The module builder does not have this information because it uses the specific module config. The information about the hook configuration is available only to the host execution plan which is processed in the prebid core. Thus meaning, that the module has no other way to know if some hooks were configured on start up.

But in the README we are highlighting the fact that the module requires both hooks.

So, the final version of this part of code would look:

```
// If the entrypoint hook was not configured, return the result without any changes
	if ctx.ModuleContext == nil {
		return result, hookexecution.NewFailure("entrypoint hook was not configured")
	}
```